src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,92 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
+    }
+};
+
+// The 2 formats also cover "%Y-%m-%d". Timezones require local_time_input_facet AND local_date_time
+// see https://www.boost.org/doc/libs/1_75_0/doc/html/date_time/date_time_io.html.
+// We strive to use smart pointers where possible. In this case this is not possible
+// but not a huge deal as these objects are statically allocated.
+const static auto formats = {
+    std::locale(std::locale::classic(),
+                new boost::local_time::local_time_input_facet("%Y-%m-%dT%H:%M:%s%ZP")),
+    std::locale(std::locale::classic(),
+                new boost::local_time::local_time_input_facet("%Y-%m-%d %H:%M:%s%ZP")),
+};
+const static boost::posix_time::ptime epoch(boost::gregorian::date(1970, 1, 1));
+const static boost::posix_time::ptime max_date(boost::gregorian::date(2030, 1, 1));
+
+static std::chrono::milliseconds parseDateAsMillis(
+    const std::string& datetime, const boost::posix_time::ptime& default_time = epoch) {
+    boost::posix_time::ptime pt;
+    auto millis = 0LL;
+
+    // Empty datetime is 0.
+    if (!datetime.empty()) {
+        for (const auto& format : formats) {
+            std::istringstream is{datetime};
+            is.imbue(format);
+            boost::local_time::local_date_time ldt{boost::local_time::not_a_date_time};
```

Comment:
is ldt and pt from date time examples? We try to avoid abbreviations, but I accept that things could get big here if we don't.

Reply:
yeah, they are from the examples, I'll remove the abbreviations.